core/src/components/chip/chip.tsx

```diff
@@ -37,21 +37,37 @@ export class Chip implements ComponentInterface {
   @Prop() disabled = false;
 
   /**
-   * Define the Chip corner shape, when using the Ionic Theme.
+   * Set to `"soft"` for a chip with slightly rounded corners, `"round"` for a chip with fully
+   * rounded corners, or `"rectangular"` for a chip without rounded corners.
+   * Defaults to `"round"` for the `"ionic"` theme and `"soft"` for all other themes.
    */
-  @Prop() shape?: 'round' | 'rectangular';
+  @Prop() shape?: 'soft' | 'round' | 'rectangular';
 
-  render() {
+  /**
+   * Set the shape based on the theme
+   */
+  private getShape(): string {
+    const theme = getIonTheme(this);
     const { shape } = this;
+
+    if (shape === undefined) {
+      return theme === 'ionic' ? 'round' : 'soft';
+    }
+
+    return shape;
+  }
+
+  render() {
     const theme = getIonTheme(this);
 
+    const shape = this.getShape();
+
     return (
       <Host
         aria-disabled={this.disabled ? 'true' : null}
         class={createColorClasses(this.color, {
           [theme]: true,
-          // TODO(FW-6120): remove the theme==='ionic' when we add support for the `ios` and `md` modes.
-          [`chip-${shape}`]: theme === 'ionic' && shape !== undefined,
+          [`chip-${shape}`]: shape !== undefined,
```

Comment:
Can shape ever be `undefined` based on the returned value from `getShape()`? 
```suggestion
          [`chip-${shape}`]: true
```

Reply:
Fixed in https://github.com/ionic-team/ionic-framework/pull/29375/commits/dcf493be18aa2d0cf51111c4a7b20ac964756f2f